zyte_common_items/pipelines.py

```diff
@@ -53,3 +58,44 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def init_thresholds(self, spider):
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = load_object(item) if isinstance(item, str) else item
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
+        return self.thresholds.get(type(item), DEFAULT_ITEM_PROBABILITY_THRESHOLD)
+
+    def get_item_name(self, item):
+        return item.__class__.__name__.lower()
+
+    def process_item(self, item, spider):
+        threshold = self.get_threshold(item, spider)
+
+        self.stats.inc_value("item/crawl/total", spider=spider)
```

Comment:
I don’t like this stat, at least not the naming. It sounds too generic, not specific to this pipeline, but it is counting the items that this pipeline sees, i.e. it is not counting items dropped by earlier item pipelines.

Maybe a prefix like `drop_low_probability_item_pipeline` would make sense, to make it clear where this setting comes from. Or maybe `probability_item_pipeline` if we rename the pipeline accordingly. So we could have `probability_item_pipeline/input_items` here (and `probability_item_pipeline/dropped_items/<type>` below; not sure if keeping the current e-commerce stat name is worth it).

Reply:
Ah, ok, if we are not going to keep e-commerce naming - yes, `probability_item_pipeline` sounds clear, might be even add word `drop` like `probability_drop_item_pipeline`.
And in this case 
probability_drop_item_pipeline/input_items/...
probability_drop_item_pipeline/dropped_items/...
And this `drop` directly adds the idea of this pipeline.

Reply:
but `probability_item_pipeline` is also good :)

Reply:
What about having the following?

- `low_proba_filter/processed`
- `low_proba_filter/dropped/Product`
- `low_proba_filter/dropped/Article`

It's short yet informative enough. I think _proba_ has a intuitive enough meaning similar to what `i` means in a for loop.

Reply:
Thanks @BurnzZ, but there is also another idea to use name of pipeline as a prefix to make this clear where the stats come from. Some suggestions [here](https://github.com/zytedata/zyte-common-items/pull/91#discussion_r1568603494) and [here](https://github.com/zytedata/zyte-common-items/pull/91#discussion_r1567791845)

Reply:
For the record, proba sounds weird to me. I would expect prob or probab when shortening probability.

Reply:
proba is prety common in ML community, e.g. scikit-learn uses it as a convention in method and argument names

Reply:
After our standup call, here's what we have agreed on:

```
drop_low_probability_item/processed
drop_low_probability_item/processed/Article
drop_low_probability_item/processed/Product
drop_low_probability_item/kept
drop_low_probability_item/kept/Article
drop_low_probability_item/kept/Product
drop_low_probability_item/dropped
drop_low_probability_item/dropped/Article
drop_low_probability_item/dropped/Product
```